tikv-client/src/main/java/com/pingcap/tikv/types/EnumType.java

```diff
@@ -103,7 +115,17 @@ protected Object decodeNotNull(int flag, CodecDataInput cdi) {
   /** {@inheritDoc} Enum is encoded as unsigned int64 with its 0-based value. */
   @Override
   protected void encodeKey(CodecDataOutput cdo, Object value) {
-    long longVal = Converter.convertToLong(value);
+    requireNonNull(value, "val is null");
+    long longVal = -1;
+    if (value instanceof Number) {
+      longVal = ((Number) value).longValue();
+    } else if (value instanceof String) {
+      longVal = parseEnumName((String)value);
+    }
```

Comment:
maybe add `else` here and throw an error

Reply:
revert the change